### PR TITLE
Allow to inject info extension data as beans

### LINKED ISSUE #32994: Quarkus-info extension could provide beans available to applications
### Description

In our applicaiton we use `git-commit-id-maven-plugin` to create a git.properties, which we add to the build via `quarkus.config.locations`

I think it would be good if we could remove that and use quarkus-info and inject or a composite bean or multiple bean for; JavaInfo, GitInfo, BuildInfo, OsInfo, which could then be use through an application. This way we could easily customize also the format we want and/or need via a custom endpoint, or not use an endpoint at all but have it available for logging or any other purposes.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Closes: #32994

### CODE REVIEW

File: extensions/info/deployment/src/test/java/io/quarkus/info/deployment/EnabledInfoTest.java [hunk 1]
```diff
@@ -4,10 +4,17 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (gastaldi) [inline extensions/info/deployment/src/test/java/io/quarkus/info/deployment/EnabledInfoTest.java:7]: Better avoid the star imports

**Review by gastaldi — APPROVED** [review]
LGTM, added a minor observation

File: extensions/info/deployment/src/test/java/io/quarkus/info/deployment/EnabledInfoTest.java (hunk 1 above)
Comment (geoand) [reply extensions/info/deployment/src/test/java/io/quarkus/info/deployment/EnabledInfoTest.java]: Thanks for catching it~

File: extensions/info/runtime/src/main/java/io/quarkus/info/OsInfo.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package io.quarkus.info;
+
+public interface OsInfo {
+
+    String name();
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/info/runtime/src/main/java/io/quarkus/info/OsInfo.java:5]: In the distant future 😁

**Pull request merged**